main.py

```diff
@@ -104,47 +119,55 @@ def main():
             print("\nWould you like to see the travel route in Google Maps?(y/n)")
             question = input()
             if question.lower() == "y":
-                webbrowser.open(construct_google_map_path(travel_df))
+                webbrowser.open(
+                    construct_google_map_path(
+                        travel_df, geocodes_df, start_lat, start_lon
+                    )
+                )
                 exit()
             else:
                 exit()
 
 
-def display_travel_route(travel_df, distance_to_start, total_distance):
+def generate_travel_route(
+    travel_df,
+    geocodes_df,
+    breweries_df,
+    start_lat,
+    start_lon,
+    distance_to_start,
+    total_distance,
+):
     """
     Format a detailed list of breweries visited on the travel route
     """
     s = ""
     number_of_breweries = travel_df["brewery_id"].count()
     s += f"Found {number_of_breweries} beer factories:\n"
     s += f"-> HOME: {start_lat} {start_lon}\n"
-    for row in travel_df.index:
-        br_id = travel_df.loc[row]["brewery_id"]
-        lat, lon = geocodes_df.loc[br_id, "latitude":"longitude"]
-        name = breweries_df.loc[br_id]["name"]
-        distance = travel_df.loc[row]["distance"]
+    for row in travel_df.itertuples():
+        lat, lon = geocodes_df.loc[row.brewery_id, "latitude":"longitude"]
+        name = breweries_df.loc[row.brewery_id]["name"]
         if len(name) > 22:
             name = name[:22] + "..."
-        s += f"-> [{br_id}] {name}: {lat} {lon} Distance: {distance} km.\n"
-
+        s += f"-> [{row.brewery_id}] {name}: {lat} {lon} Distance: {row.distance} km.\n"
     s += f"<- HOME: {start_lat}, {start_lon} Distance: {distance_to_start} km.\n\n"
     s += f"Total distance: {total_distance} km.\n"
     return s
 
 
-def display_beer_list(travel_df):
+def generate_beer_list(travel_df, beers_df):
     """
     Format a detailed list of beer types collected on the travel route
     """
     s = ""
     s += f"Collected {count_beer(travel_df)} beer types:\n"
-    for row in travel_df.index:
-        br_id = travel_df.loc[row]["brewery_id"]
-        if type(beers_df.loc[br_id]["name"]) == str:
-            name = beers_df.loc[br_id]["name"]
+    for row in travel_df.itertuples():
+        if type(beers_df.loc[row.brewery_id]["name"]) == str:
```

Comment:
Why is this `type` needed?

Reply:
Some breweries only have 1 type of beer, so, when this line is called:
`beers_df.loc[row.brewery_id]["name"]`
it returns a String and not a DataFrame. So, if it's a string, there's no need to iterate over it.
On the other hand, when brewery has many types of beer, that line returns a DataFrame and you have to iterate over it to collect all the names.

Reply:
Can you produce a code example that demonstrates how this works?
I would like to see how its string one time and other time its dataframe

Reply:
Code example, when printing values produced by the mentioned line of code.
DataFrame:
```
brewery_id
961           Porter IV
961    Koff Special III
Name: name, dtype: object
```
String:
`Porter`

Whole:
```
brewery_id
961           Porter IV
961    Koff Special III
Name: name, dtype: object
Porter
```

Reply:
Fixed